components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java

```diff
@@ -206,17 +246,180 @@ public void testHandleClaimsForOAuthAuthzReqMessageContext() throws Exception {
+
+
+        ApplicationManagementService mockApplicationManagementService = mock(ApplicationManagementService.class);
+        mockStatic(OAuth2ServiceComponentHolder.class);
+        when(OAuth2ServiceComponentHolder.getApplicationMgtService()).thenReturn(mockApplicationManagementService);
+
+        when(mockApplicationManagementService.getServiceProviderNameByClientId(anyString(), anyString(), anyString()))
+                .thenReturn(SAMPLE_SERVICE_PROVIDER);
+        ServiceProvider serviceProvider = mock(ServiceProvider.class);
+        when(mockApplicationManagementService.getApplicationExcludingFileBasedSPs(anyString(), anyString()))
+                .thenReturn(serviceProvider);
+
+        ClaimConfig claimConfig = mock(ClaimConfig.class);
+        when(serviceProvider.getClaimConfig()).thenReturn(claimConfig);
+
+        ClaimMapping claimMap1 = ClaimMapping.build("http://www.wso2.org/claims/email", "http://www.abc.com/claims/email", "sample@abc.com", false);
+        ClaimMapping claimMap2 = ClaimMapping.build("http://www.wso2.org/claims/username", "http://www.abc.com/claims/username", "user123", true);
+
+        ClaimMapping[] requestedLocalClaimMap = {claimMap1, claimMap2};
+        when(claimConfig.getClaimMappings()).thenReturn(requestedLocalClaimMap);
+
+        org.wso2.carbon.user.core.UserRealm sampleUserRealm = mock(org.wso2.carbon.user.core.UserRealm.class);
+        mockStatic(IdentityTenantUtil.class);
+        when(IdentityTenantUtil.getRealm(anyString(), anyString())).thenReturn(sampleUserRealm);
+
+        Map<String, String> spToLocalClaimMappings = new HashMap<>();
+        mockStatic(ClaimMetadataHandler.class);
+        ClaimMetadataHandler claimMetadataHandler = mock(ClaimMetadataHandler.class);
+
+        when(ClaimMetadataHandler.getInstance()).thenReturn(claimMetadataHandler);
+        when(claimMetadataHandler.getMappingsMapFromOtherDialectToCarbon(anyString(), anySet(), anyString(),
+                anyBoolean())).thenReturn(spToLocalClaimMappings);
+
+        Map<String, String> userClaims = new HashMap<>();
+        mockStatic(UserCoreUtil.class);
+        when(UserCoreUtil.addDomainToName(anyString(), anyString())).thenReturn("user1234@abc.com");
+
+        userStoreManager = mock(UserStoreManager.class);
+        when(sampleUserRealm.getUserStoreManager()).thenReturn(userStoreManager);
+        when(userStoreManager.getUserClaimValues(anyString(), any(String[].class), anyString())).thenReturn(userClaims);
+
+        RealmConfiguration realmConfiguration = mock(RealmConfiguration.class);
+        when(userStoreManager.getSecondaryUserStoreManager(anyString())).thenReturn(userStoreManager);
+        when(userStoreManager.getRealmConfiguration()).thenReturn(realmConfiguration);
+        when(realmConfiguration.getUserStoreProperty(IdentityCoreConstants.MULTI_ATTRIBUTE_SEPARATOR))
+                .thenReturn(":");
+        startTenantFlowProcess();
```

Comment:
If you are starting a tenant flow, you need to end it as well. Lets do it within a try finally block